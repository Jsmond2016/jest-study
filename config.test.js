const { generateConfig, generateConfig2 } = require('./config')

//使用命令 jest --updateSnapshot 更新快照  https://jestjs.io/docs/cli#--updatesnapshot
// 使用场景：ui 组件

test('测试配置文件的快照-generateConfig', () => {
  expect(generateConfig()).toMatchSnapshot();
})
test('测试配置文件的快照-generateConfig', () => {
  expect(generateConfig2()).toMatchSnapshot({
    dynamicTime: expect.any(Date)
  });
})

// toMatchInlineSnapshot 将快照放在代码中
test('测试配置文件的快照-toMatchInlineSnapshot-generateConfig', () => {
  expect(generateConfig2()).toMatchInlineSnapshot({
  dynamicTime: expect.any(Date) }, `
Object {
  "domain": "localhost",
  "dynamicTime": Any<Date>,
  "port": 8080,
  "server": "http://localhost",
  "time": "2022",
}
`);
})