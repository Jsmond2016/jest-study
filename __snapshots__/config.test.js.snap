// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试配置文件的快照-generateConfig 1`] = `
Object {
  "domain": "localhost",
  "port": 8080,
  "server": "http://localhost",
  "time": "2022",
}
`;

exports[`测试配置文件的快照-generateConfig 2`] = `
Object {
  "domain": "localhost",
  "dynamicTime": Any<Date>,
  "port": 8080,
  "server": "http://localhost",
  "time": "2022",
}
`;
